browser-extension/mv2/src/background/background.js

```diff
@@ -264,13 +269,50 @@ BG.Methods.modifyHeaders = function (originalHeaders, headersTarget, details) {
     ruleType,
     rulePairs,
     rulePair,
-    isRuleApplied = false,
+    hasModifiedHeaders = false,
     modifications,
     modification,
     url = details.url,
     mainFrameUrl = BG.Methods.getMainFrameUrl(details),
     enabledRules = BG.Methods.getEnabledRules();
 
+  if (headersTarget === RQ.HEADERS_TARGET.REQUEST) {
+    RQ.IGNORED_HEADERS_ON_REDIRECT.forEach((headerName) => {
```

Comment:
I think we should structure this code into separate functions like this

isHeadersModified = false;
If (target === requestHeaders) {
  isHeadersModified = copyIgnoredHeadersInRedirect(originalHeaders);
} else {
  isHeadersModified = addCORSHeadersForRQCustomHeaders(originalHeaders);
}

Write Unit tests for each of these functions separately.